postgres_composite_types/__init__.py

```diff
@@ -258,10 +259,32 @@ def __init__(cls, name, bases, attrs):
         if name == 'CompositeType':
             return
 
+        cls._capture_descriptors()
+
         # Register the type on the first database connection
         connection_created.connect(receiver=cls.database_connected,
                                    dispatch_uid=cls._meta.db_type)
 
+    def _capture_descriptors(cls):
+        """ Work around for not being able to call contribute_to_class."""
```

Comment:
If we did it in the metaclass's `__new__` can we save ourselves from having to make another type to do the introspection?